tracks/python/exercises/leap/mentoring.md

```diff
@@ -1,3 +1,24 @@
+### Most common first attempt
+
+This is the most common solution by beginners to the leap year problem.
+Nudge the student towards the more concise solution by letting them know
+they can replace the 3 IF and 4 return statements with zero IF statements and a single return.
+This is done by using the logical operators OR & AND.
```

Comment:
```suggestion
This is a common solution to the leap year problem. You can nudge the student towards
a more concise solution by letting them know that they can replace the third
`if ... else` block with a single `return year % 400 == 0` statement.
```

Reply:
Why would we reverse the suggestion of not using if statements, as a mentor note?

I do agree with removing "most common solution by beginners" as we can not be sure they are beginners, and the "most common" can change over time, implying a maintenance requirement.

Reply:
Oh I misinterpreted their suggestion, I though they referred to the 3rd `IF`.

```python
def leap_year(year):
    return year % 4 == 0 and (year % 100 == 0 or year % 400 == 0)
 ```

I mean, is this actually better though? I know that after passing through the `Exercism` course myself the first time and then contributing to other projects I really had to adjust to not trying to fit everything on a single 80 character line. In terms of readability I do think you're losing somewhat here.
For this exercise it is probably fine, but by suggesting to write code like this beginners will probably think it is always "better" to write such one-liners (which I personally don't agree with).

Reply:
It does not unduly check for 400.  It is not for the sake of "a single line" or "80 characters".

It is a mentor note, and so the mentor may not always suggest this, taking into consideration where students are based on the code and conversation that is had.

But yeah, this is not because of "one line", only an "understand the language and its constructs" note, and I would say that it is is up to the student to gain the experience to know when applying this and something like this is "better".

Reply:
```suggestion
This is a common solution to the leap year problem. You can nudge the student towards
a more concise solution by letting them know that they can replace all `if` and `return`
statements with one `return` statement that concatenates all checks using `and` and `or`.
```

Reply:
@kotp Something like this then?